openedx/core/djangoapps/content_libraries/libraries_index.py

```diff
@@ -0,0 +1,159 @@
+""" Code to allow indexing content libraries """
+
+import logging
+
+from django.conf import settings
+from django.dispatch import receiver
+from elasticsearch.exceptions import ConnectionError
+from search.search_engine_base import SearchEngine
+
+from openedx.core.djangoapps.content_libraries.signals import (
+    CONTENT_LIBRARY_CREATED,
+    CONTENT_LIBRARY_UPDATED,
+    CONTENT_LIBRARY_DELETED,
+    LIBRARY_BLOCK_CREATED,
+    LIBRARY_BLOCK_UPDATED,
+    LIBRARY_BLOCK_DELETED,
+)
+from openedx.core.djangoapps.content_libraries.library_bundle import LibraryBundle
+from openedx.core.djangoapps.content_libraries.models import ContentLibrary
+from openedx.core.lib.blockstore_api import get_bundle
+
+log = logging.getLogger(__name__)
+
+MAX_SIZE = 10000  # 10000 is the maximum records elastic is able to return in a single result. Defaults to 10.
+
+
+class LibraryNotIndexedException(Exception):
+    """
+    Library supplied wasn't indexed in ElasticSearch
+    """
+    pass
+
+
+class ContentLibraryIndexer:
+    """
+    Class to perform indexing for blockstore-based content libraries
+    """
+
+    INDEX_NAME = "content_library_index"
+    LIBRARY_DOCUMENT_TYPE = "content_library"
+
+    @classmethod
+    def index_libraries(cls, library_keys):
+        """
+        Index the specified libraries. If they already exist, replace them with new ones.
+        """
+        from openedx.core.djangoapps.content_libraries.api import DRAFT_NAME
```

Comment:
I like avoiding these type of imports. Consider factoring `DRAFT_NAME` out into constants file.